app/models/recipe_ingredient.rb

```diff
@@ -1,3 +1,13 @@
 class RecipeIngredient < ApplicationRecord
 
+  belongs_to :recipe
+  belongs_to :ingredient
+
+  def self.find_recipe_ingredients(recipe)
+    RecipeIngredient.joins(:ingredient).where(recipe_id: recipe).select("ingredients.*")
+  end
+
+  def self.find_recipe_ingredients_cost(recipe)
+    RecipeIngredient.joins(:ingredient).where(recipe_id: recipe).select("ingredients.*").sum(:cost)
+  end
 end
```

Comment:
I talked about these methods earlier, but again, these are not necessary at all. For future reference, I would add the schema to a db diagramming website and take a look at the tables to determine which are necessary for any query. It could eliminate situations like this one.